# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/ConcurrentDelazification.cpp
# Commit: f10838447589
# Full Hash: f10838447589dc565d7d12d39f6e00ae00f98175
# Author: Tooru Fujisawa <arai_a@mac.com>
# Date: 2024-12-05 09:58:04
# Regressor Bug: 1933681
# File Overlap Count: 1
# Description:
#   Bug 1933681 - Part 7: Move storeDelazification call into CompileLazyFunctionToStencilMaybeInstantiate. r=nbp
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D231022
# ==============================================================================

diff -r adb4d60ce68c -r f10838447589 js/src/vm/ConcurrentDelazification.cpp
--- a/js/src/vm/ConcurrentDelazification.cpp	Wed Dec 04 23:44:35 2024 +0000
+++ b/js/src/vm/ConcurrentDelazification.cpp	Wed Dec 04 23:44:35 2024 +0000
@@ -220,7 +220,7 @@
       isInterrupted_ = false;
       break;
     }
-    RefPtr<CompilationStencil> innerStencil;
+    const CompilationStencil* innerStencil;
     ScriptIndex scriptIndex = strategy_->next();
     {
       BorrowingCompilationStencil borrow(merger_.getResult());
@@ -244,13 +244,10 @@
       }
     }
 
-    const CompilationStencil* innerStencilPtr =
-        stencils_->storeDelazification(std::move(innerStencil));
-
     // We are merging the delazification now, while this could be post-poned
     // until we have to look at inner functions, this is simpler to do it now
     // than querying the cache for every enclosing script.
-    if (!merger_.addDelazification(&fc_, *innerStencilPtr)) {
+    if (!merger_.addDelazification(&fc_, *innerStencil)) {
       strategy_->clear();
       return false;
     }
